patches/@react-pdf+renderer+2.0.0-beta.14.patch

```diff
@@ -0,0 +1,136 @@
+diff --git a/node_modules/@react-pdf/renderer/lib/react-pdf.browser.cjs.js b/node_modules/@react-pdf/renderer/lib/react-pdf.browser.cjs.js
+index e164b4b..d8693ea 100644
+--- a/node_modules/@react-pdf/renderer/lib/react-pdf.browser.cjs.js
++++ b/node_modules/@react-pdf/renderer/lib/react-pdf.browser.cjs.js
+@@ -153,7 +153,7 @@ var createRenderer = function createRenderer(_ref) {
+   });
+ };
+ 
+-var version = "2.0.0-beta.13";
++var version = "2.0.0-beta.14";
+ 
+ var fontStore = new FontStore();
+ 
+@@ -358,6 +358,10 @@ var InternalBlobProvider = /*#__PURE__*/function (_React$PureComponent) {
+     };
+ 
+     _this.queueDocumentRender = function () {
++      _this.setState({
++        loading: true
++      });
++
+       _this.renderQueue.splice(0, _this.renderQueue.length, function () {
+         return _this.state.error ? Promise.resolve() : _this.instance.toBlob();
+       });
+@@ -377,7 +381,8 @@ var InternalBlobProvider = /*#__PURE__*/function (_React$PureComponent) {
+       _this.setState({
+         blob: blob,
+         url: URL.createObjectURL(blob),
+-        loading: false
++        loading: false,
++        error: null
+       }, function () {
+         return URL.revokeObjectURL(oldBlobUrl);
+       });
+@@ -397,12 +402,18 @@ var InternalBlobProvider = /*#__PURE__*/function (_React$PureComponent) {
+     this.renderQueue.on('success', this.onRenderSuccessful);
+   };
+ 
+-  _proto.componentDidUpdate = function componentDidUpdate() {
+-    this.instance.updateContainer(this.props.document);
++  _proto.componentDidUpdate = function componentDidUpdate(prevProps) {
++    if (prevProps.document !== this.props.document) {
++      this.instance.updateContainer(this.props.document);
++    }
+   };
+ 
+   _proto.componentWillUnmount = function componentWillUnmount() {
+     this.renderQueue.end();
++
```

Comment:
This file makes me more nervous the larger it gets as we may diverge in months time. Can we have a fork of react-pdf/renderer and open a pull request?

Reply:
Yes I made PRs today https://github.com/diegomura/react-pdf/pull/995 and https://github.com/diegomura/react-pdf/pull/996.

Forking would be a pain because it would mean publishing all the linked packages from the v2 monorepo. I think the .patch is ok, but if the PRs are not merged in the next release we should re-visit.

Reply:
We can depend upon github forks so it doesn't have to be published as a fork; but sure, this is fine and hopefully they'll merge and publish.

Reply:
Yeah unfortunately most front-end packages don't work from github forks because what is published to npm is built code. Node coding is much easier in this respect! Hence why I lean on patch-package for front-end stuff as the least-friction option. For mapeo-mobile I mantain a [README](https://github.com/digidem/mapeo-mobile/blob/develop/patches/README.md) in the patches folder to track what the changes are in there for. I have found it easier to maintain than publishing forks on npm.

Reply:
ah thanks for that, that's helpful!